pandas/tests/test_reshape.py

```diff
@@ -56,6 +56,38 @@ def test_value_vars(self):
                               columns=['id1', 'id2', 'variable', 'value'])
         tm.assert_frame_equal(result4, expected4)
 
+    def test_value_vars_types(self):
+        # GH 15348
+        expected = DataFrame({'id1': self.df['id1'].tolist() * 2,
+                              'id2': self.df['id2'].tolist() * 2,
+                              'variable': ['A'] * 10 + ['B'] * 10,
+                              'value': (self.df['A'].tolist() +
+                                        self.df['B'].tolist())},
+                             columns=['id1', 'id2', 'variable', 'value'])
+
+        for type_ in (tuple, list, np.array):
+            result = melt(self.df, id_vars=['id1', 'id2'],
+                          value_vars=type_(('A', 'B')))
+            tm.assert_frame_equal(result, expected)
+
+    def test_id_and_value_vars_types_with_multiindex(self):
+        expected = DataFrame({
+            ('A', 'a'): self.df1[('A', 'a')],
+            'CAP': ['B'] * len(self.df1),
+            'low': ['b'] * len(self.df1),
+            'value': self.df1[('B', 'b')],
+        }, columns=[('A', 'a'), 'CAP', 'low', 'value'])
+
+        for id_vars in ([('A', 'a')], ('A', 'a')):
```

Comment:
separate this in to 2 parts, the success part (e.g. both lists), and the failing parts (the other 3 possibilites)